celery_unique/core.py

```diff
@@ -0,0 +1,199 @@
+            identify this task. The function will take receive the same args
+            and kwargs the task is passed.
+        :param UniquenessBackend backend: A backend to use to cache queued
+            tasks and to determine is a task is unique or not.
+        """
+        should_handle_as_unique_task = self._handle_as_unique(options)
+
+        if should_handle_as_unique_task:
+            # Generate the unique key and revoke any task that shares the same
+            # key (if one exists)
+            key = self.make_key(args, kwargs)
+            self._revoke_extant_task(key)
+
+        # Pass the task along to Celery for publishing and intercept the
+        # AsyncResult return value
+        rv = super(UniqueTaskMixin, self).apply_async(args, kwargs, task_id, **options)
+
+        if should_handle_as_unique_task:
+            # Inform the backend of this tasks.
+            # The new record will be given a TTL that allows it to expire
+            # (approximately) at the same time that the task is executed.
+            ttl = self._make_ttl_for_unique_task_record(options)
+            self.unique_backend.create_task_record(key, rv.task_id, ttl)
+
+        return rv
+
+    def _handle_as_unique(self, options):
+        """
+        Determines if a task should be handles as unique.
+
+        :param dict options: The options dict passed to `apply_async`.
+        """
+        return (
+            callable(self.unique_key)
+            and ('eta' in options or 'countdown' in options)
+            and self.unique_backend is not None
+        )
+
+    def _revoke_extant_task(self, key):
+        """
+        Removes an extant task
+
+        Not that this removed the key from the store regardless of whether the
+        task is still queued or not.
+        """
+        task_id = self.unique_backend.get_task_id(key)
+        if task_id is not None:
+            self.app.AsyncResult(task_id).revoke()
+            self.unique_backend.revoke_extant_task(key)
+
+    def make_key(self, callback_args, callback_kwargs):
+        """
+        Creates a key used to identify the task's unique configuration in Redis.
+
+        Note: All positional arguments and/or keyword arguments sent to the
+        task are applied identically to the task's bound `unique_key` callable.
+
+        :param callback_args: The positional arguments which will be passed to
+            the task when it executes @type callback_args: list | tuple
+        :param callback_kwargs: The keyword arguments which will be passed to
+            the task when it executes @type callback_kwargs: dict
+
+        :return: The key which will be used to find any extant version of this
+        task which, if found, will by revoked.  Keys are built by using three
+        colon-delimited components::
+
+            1. A global prefix used to identify that the key/value pair in
+                the backend was created to track
```

Comment:
I think this line's break occurs too early so it looks a bit awkward.  Suggest this instead:
```
1. A global prefix used to identify that the key/value pair in
    the backend was created to track a unique Celery task 
    (by default, this is "celery_unique")
```